File: backend/src/main/java/com/ssafy/hellingers/service/UserService.java
package com.ssafy.hellingers.service;

import com.ssafy.hellingers.model.Role;
import com.ssafy.hellingers.model.User;

import java.util.List;
import java.util.Map;
import java.util.Optional;


public interface UserService
{
    User saveUser(User user);

//    User updateUser(User user);

    Optional<User> login(String email, String password) throws Exception ;

    User changeRole(Role newRole, String username);

    Optional<User> findByEmail(String email);

    Optional<User> findByNickname(String nickname);

//    Optional<User> findById(Long id);

    User deleteUser(Long id);

    List<User> findAllUsers();
}

File: frontend/src/services/user.service.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/';

class UserService {

    login(user) {
        console.log(user)
        // const header = {
        //     authorization: 'Basic ' + btoa(user.username + ':' + user.password)
        // };
        // let obj = {user.email, user.password}
        // console.log(obj)
        return axios.post(API_URL + 'login', user);
    }

    // logOut() {
    //     return axios.post(API_URL + 'logout', {});
    // }

    signUp(user) {
      console.log('레지')
      return axios.post(API_URL + 'signup', user);
    }

    changeRole(username, role) {
        return axios.put(API_URL + username + '/change/' + role, {});
    }

}

export default new UserService();
